### PR TITLE
add: inclusion proof and dc sidecar verification

### PR DESCRIPTION
**Pull request by agnxsh**
This PR adds the following specs:
1. https://github.com/ethereum/consensus-specs/blob/v1.5.0-alpha.6/specs/_features/eip7594/p2p-interface.md#verify_data_column_sidecar_kzg_proofs
2. https://github.com/ethereum/consensus-specs/blob/v1.5.0-alpha.6/specs/_features/eip7594/p2p-interface.md#verify_data_column_sidecar_inclusion_proof

### CODE REVIEW

File: beacon_chain/spec/eip7594_helpers.nim [hunk 1]
```diff
@@ -306,4 +306,57 @@ func get_extended_sample_count*(samples_per_slot: int,
...
+
+# https://github.com/ethereum/consensus-specs/blob/v1.5.0-alpha.6/specs/_features/eip7594/p2p-interface.md#verify_data_column_sidecar_inclusion_proof
+proc verify_data_column_sidecar_inclusion_proof*(sidecar: DataColumnSidecar):
+                                                 Result[void, string] =
+  ## Verify if the given KZG Commitments are in included
+  ## in the beacon block or not
+  let gindex = 
+    KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH_GINDEX.GeneralizedIndex
+  if not is_valid_merkle_branch(
+    hash_tree_root(sidecar.kzg_commitments),
```
Comment (tersec) [inline beacon_chain/spec/eip7594_helpers.nim]: To disambiguate the `if` condition from the body while keeping the `body` two spaces indent, can indent the `if` condition 4 spaces:
```nim
if not ivmb(
    line1,
    line2,
    line3):
  body
```

File: beacon_chain/spec/eip7594_helpers.nim [hunk 2]
```diff
@@ -306,4 +306,57 @@ func get_extended_sample_count*(samples_per_slot: int,
...
+  ## Verify if the KZG Proofs consisting in the `DataColumnSidecar`
+  ## is valid or not.
+  
+  # Check if the data column sidecar index < NUMBER_OF_COLUMNS
+  if not (sidecar.index < NUMBER_OF_COLUMNS):
+    return err("Data column sidecar index exceeds the NUMBER_OF_COLUMNS")
+
+  # Check is the sidecar column length = sidecar.kzg_commitments length
+  # and sidecar.kzg_commitments length = sidecar.kzg_proofs length
+
```
Comment (tersec) [inline beacon_chain/spec/eip7594_helpers.nim]: there's an extra line here? seems inconsistent with spacing in rest of function

File: beacon_chain/spec/eip7594_helpers.nim [hunk 3]
```diff
@@ -306,4 +306,57 @@ func get_extended_sample_count*(samples_per_slot: int,
...
+    KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH_GINDEX.GeneralizedIndex
+  if not is_valid_merkle_branch(
+    hash_tree_root(sidecar.kzg_commitments),
+    sidecar.kzg_commitments_inclusion_proof,
+    KZG_COMMITMENTS_INCLUSION_PROOF_DEPTH.int,
+    get_subtree_index(gindex),
+    sidecar.signed_block_header.message.body_root):
+    
+    return err("DataColumnSidecar: Inclusion proof is invalid")
+  ok()
```
Comment (tersec) [inline beacon_chain/spec/eip7594_helpers.nim:327]: `verify_data_column_sidecar_kzg_proofs` has an extra line before `ok()` and this doesn't, probably good to be consistent about it

File: beacon_chain/spec/eip7594_helpers.nim [hunk 4]
```diff
@@ -306,4 +306,57 @@ func get_extended_sample_count*(samples_per_slot: int,
         worstCaseConditionCount, i) <= falsePositiveThreshold:
       return i
 
-  NUMBER_OF_COLUMNS
\ No newline at end of file
+  NUMBER_OF_COLUMNS
+
+# https://github.com/ethereum/consensus-specs/blob/v1.5.0-alpha.6/specs/_features/eip7594/p2p-interface.md#verify_data_column_sidecar_inclusion_proof
+proc verify_data_column_sidecar_inclusion_proof*(sidecar: DataColumnSidecar):
+                                                 Result[void, string] =
```
Comment (tersec) [inline beacon_chain/spec/eip7594_helpers.nim]: `cstring` would suffice here and `verify_data_column_sidecar_kzg_proofs` and is more efficient (no allocations, garbage collection, et cetera)

File: beacon_chain/spec/eip7594_helpers.nim (hunk 1 above)
Comment (agnxsh) [reply beacon_chain/spec/eip7594_helpers.nim]: done

File: beacon_chain/spec/eip7594_helpers.nim (hunk 3 above)
Comment (agnxsh) [reply beacon_chain/spec/eip7594_helpers.nim:327]: done

File: beacon_chain/spec/eip7594_helpers.nim (hunk 2 above)
Comment (agnxsh) [reply beacon_chain/spec/eip7594_helpers.nim]: done

File: beacon_chain/spec/eip7594_helpers.nim (hunk 4 above)
Comment (agnxsh) [reply beacon_chain/spec/eip7594_helpers.nim]: done

**Pull request merged**